src/pelib/PeLibAux.cpp

```diff
@@ -135,6 +139,23 @@ namespace PeLib
 		return (ByteSize >> PELIB_PAGE_SIZE_SHIFT) + ((ByteSize & (PELIB_PAGE_SIZE - 1)) != 0);
 	}
 
+	bool pelibIsBufferZeroed(const void * buffer, size_t length)
+	{
+		const uint32_t * buffer32 = reinterpret_cast<const uint32_t *>(buffer);
```

Comment:
The implementation seems unnecessarily complicated and I had a hard time understanding it. Couldn't we replace it with this tiny template (in `include/retdec/pelib/PeLibAux.h`)?
```cpp
template<typename T>
bool pelibIsBufferZeroed(const std::vector<T>& buffer)
{
    return std::all_of(buffer.begin(), buffer.end(), [](T item) { return item == 0; });
}
```
The prototype of the function is simpler, we no longer need to separately pass the buffer and its size, there are no reinterpret casts, and the code is much more readable (it just checks that all items in the buffer are zero, just like you would expect).

Or does the implementation do something other than checking if the array contains only zeros?